internal/infrastructure/storage/db/badger/trade_repository_impl.go

```diff
@@ -210,34 +210,45 @@ func (t tradeRepositoryImpl) getTrade(
 	ctx context.Context,
 	ID uuid.UUID,
 ) (*domain.Trade, error) {
-	var trade domain.Trade
+	var trades []domain.Trade
 	var err error
+	query := badgerhold.Where("ID").Eq(ID)
 	if ctx.Value("tx") != nil {
 		tx := ctx.Value("tx").(*badger.Txn)
-		err = t.store.TxGet(tx, ID, &trade)
+		err = t.store.TxFind(tx, &trades, query)
 	} else {
-		err = t.store.Get(ID, &trade)
+		err = t.store.Find(&trades, query)
 	}
 	if err != nil {
-		if err == badgerhold.ErrNotFound {
-			return nil, nil
-		}
 		return nil, err
 	}
 
-	return &trade, nil
+	if len(trades) <= 0 {
+		return nil, nil
+	}
+
+	return &trades[0], nil
 }
 
 func (t tradeRepositoryImpl) updateTrade(
 	ctx context.Context,
 	ID uuid.UUID,
 	trade domain.Trade,
 ) error {
+	query := badgerhold.Where("ID").Eq(ID)
 	if ctx.Value("tx") != nil {
 		tx := ctx.Value("tx").(*badger.Txn)
-		return t.store.TxUpdate(tx, ID, trade)
+		return t.store.TxUpdateMatching(tx, &domain.Trade{}, query, func(record interface{}) error {
+			tr := record.(*domain.Trade)
```

Comment:
seem panic prone. shall we have a domain method?

Reply:
no this is not panic prone, the 2 argument of the function (`&domain.Trade{}`) defines the concrete type of `record`, therefore it's safe to cast without checking the type. We already do the same in the unspent repo btw.

> shall we have a domain method?

No, because this is related to the badger impl. The domain must not depend on or know anything about badger or any other concrete implementation.